src/client/ContentManagement.js

```diff
@@ -616,13 +621,55 @@ exports.CreateContentObject = async function({libraryId, objectId, options={}})
  *
  * @returns {Promise<Object>} - Response containing the object ID and write token of the draft
  */
-exports.CopyContentObject = async function({libraryId, originalVersionHash, options={}}) {
+exports.CopyContentObject = async function({libraryId, originalVersionHash, options={}, originalObject=null}) {
   ValidateLibrary(libraryId);
   ValidateVersion(originalVersionHash);
 
   options.copy_from = originalVersionHash;
 
-  return await this.CreateContentObject({libraryId, options});
+  if(!originalObject) {
+    return await this.CreateAndFinalizeContentObject({libraryId, options});
+  }
+
+  ValidateObject(originalObject.id);
+
+  const callback = async ({objectId, writeToken}) => {
+    const kmsCapKey = Object.keys(originalObject).find(key => key.startsWith("eluv.caps.ikms"));
+    const userCapKey = Object.keys(originalObject).find(key => key.startsWith("eluv.caps.iusr"));
+
+    if(userCapKey) {
+      const userConkKey = await this.Crypto.DecryptCap(originalObject[userCapKey], this.signer.signingKey.privateKey);
+      userConkKey.qid = objectId;
+
+      await this.DeleteMetadata({
+        libraryId,
+        objectId,
+        writeToken,
+        metadataSubtree: userCapKey
+      });
+
+      this.ReplaceMetadata({
+        libraryId,
+        objectId,
+        writeToken,
+        metadataSubtree: userCapKey,
+        metadata: await this.Crypto.EncryptConk(userConkKey, this.signer.signingKey.publicKey)
+      });
+    }
+
+    if(kmsCapKey) {
+      await this.DeleteMetadata({
+        libraryId,
+        objectId,
+        writeToken,
+        metadataSubtree: kmsCapKey
+      });
+
+      await this.SetPermission({objectId, permission: originalObject.permission, writeToken});
+    }
+  };
+
+  return await this.CreateAndFinalizeContentObject({libraryId, options, callback});
```

Comment:
I think the code would look clearer if you either removed the definition of the callback method and used the usual CreateContentObject -> do stuff -> FinalizeContentObject flow, or defined the callback in the CreateAndFinalize parameters instead of separately. At the moment it's not clear that you're just doing a CreateAndFinalize until way down at the bottom of the method.

```
const {objectId, writeToken} = await this.CreateContentObject({libraryId, options});

...

return await this.FinalizeContentObject({libraryId, objectId, writeToken});
```

or

```
await this.CreateAndFinalizeContentObject({
  libraryId, 
  options,
  callback: async ({objectId, writeToken}) => {
    ...
   }
});
```